File: Sort_ParityII/sort.py
from typing import List


class Solution:
    def sortArrayByParityII(self, nums: List[int]) -> List[int]:
        sort_parity: List[int] = [0] * len(nums)

        i, j = 0, 1

        for num in nums:
            if num % 2 == 0:
                sort_parity[i] = num
                i += 2
            else:
                sort_parity[j] = num
                j += 2

        return sort_parity

File: Sort_ParityII/sort.cs
using System.Collections.Generic;

public class Solution
{
    public int[] SortArrayByParityII(int[] nums)
    {
        int[] result = new int[nums.Length];
        int i = 0,
            j = 1;

        for (int k = 0; k < nums.Length; k++)
        {
            if (nums[k] % 2 == 0)
            {
                result[i] = nums[k];
                i += 2;
            }
            else
            {
                result[j] = nums[k];
                j += 2;
            }
        }

        return result;
    }
}
